src/wp-includes/l10n.php

```diff
@@ -1321,14 +1280,18 @@ function get_translations_for_domain( $domain ) {
  *
  * @since 3.0.0
  *
- * @global MO[] $l10n
+ * @global MO[]                   $l10n                   An array of all currently loaded text domains.
+ * @global MO[]                   $l10n_unloaded          An array of all text domains that have been unloaded again.
+ * @global WP_Textdomain_Registry $wp_textdomain_registry WordPress Textdomain Registry.
  *
  * @param string $domain Text domain. Unique identifier for retrieving translated strings.
  * @return bool Whether there are translations.
  */
 function is_textdomain_loaded( $domain ) {
-	global $l10n;
-	return isset( $l10n[ $domain ] );
+	global $l10n, $l10n_unloaded, $wp_textdomain_registry;
+
+	/** @var WP_Textdomain_Registry $wp_textdomain_registry */
+	return isset( $l10n[ $domain ] ) || ( ! isset( $l10n_unloaded[ $domain ] ) && $wp_textdomain_registry->get( $domain ) );
```

Comment:
This is reason for the last failing test. @swissspidy I'm not sure I understand this change, why should a text domain in the registry now determine the text domain as loaded?

Reply:
I think it's because in my patch I tried to shove too many responsibilities into `WP_Textdomain_Registry`, when all it really needs to do is serve as a cache. This change seems to be a remnant of that, and I think can be reverted.